include/fiction/algorithms/physical_design/ortho_ordering_network.hpp

```diff
@@ -0,0 +1,633 @@
+            {
+                const auto n_s = node2pos[po];
+
+                tile<Lyt> po_tile{};
+
+                if (std::find(out_nodes.begin(), out_nodes.end(), po) != out_nodes.end())
+                {
+                    multi_out_node = true;
+                }
+
+                // determine PO orientation
+                if (is_eastern_po_orientation_available(ctn, po) && !multi_out_node)
+                {
+                    po_tile = static_cast<tile<Lyt>>(n_s);
+                    layout.resize({latest_pos.x, latest_pos.y - 1, 1});
+                }
+                else
+                {
+                    po_tile = static_cast<tile<Lyt>>(n_s);
+                    po_tile = static_cast<tile<Lyt>>(wire_south(layout, po_tile, {po_tile.x, po_tile.y + 2}));
+                    layout.resize({latest_pos.x, latest_pos.y - 1, 1});
+                }
+                // check if PO position is located at the border
+                if (layout.is_at_eastern_border({po_tile.x + 1, po_tile.y}) && !multi_out_node)
+                {
+                    ++po_tile.x;
+                    layout.create_po(n_s,
+                                     ctn.color_ntk.has_output_name(po_counter) ?
+                                         ctn.color_ntk.get_output_name(po_counter++) :
+                                         fmt::format("po{}", po_counter++),
+                                     po_tile);
+                }
+                // place PO at the border and connect it by wire segments
+                else
+                {
+                    const tile<Lyt> anker{po_tile};
+
+                    po_tile = layout.eastern_border_of(po_tile);
+
+                    layout.create_po(wire_east(layout, anker, po_tile),
+                                     ctn.color_ntk.has_output_name(po_counter) ?
+                                         ctn.color_ntk.get_output_name(po_counter++) :
+                                         fmt::format("po{}", po_counter++),
+                                     po_tile);
+                    multi_out_node = false;
+                }
+
+                out_nodes.push_back(po);
+            });
+
+        // restore possibly set signal names
+        restore_names(ctn.color_ntk, layout, node2pos);
+
+        // statistical information
+        pst.x_size    = layout.x() + 1;
+        pst.y_size    = layout.y() + 1;
+        pst.num_gates = layout.num_gates();
+        pst.num_wires = layout.num_wires();
+
+        return layout;
+    }
+
+  private:
+    input_ordering_view<mockturtle::fanout_view<mockturtle::names_view<technology_network>>> ntk;
+
+    orthogonal_physical_design_params ps;
+    orthogonal_physical_design_stats& pst;
+
+    uint32_t po_counter{0};
+};
+
+}  // namespace detail
+
+/**
+ * A modification of the orthogonal algorithm, which saves area and wire crossings by ordering PIs
+ * and coloring the affected nodes based on the relevant conditions.
```

Comment:
Can this documentation be a bit more verbose? Particularly of interest would be the input ordering network itself. How does it work, and what does it achieve?